lib/participant_profile_deduplicator.rb

```diff
@@ -48,17 +51,29 @@ def log_overview_info
     log_info("Duplicate profile: #{duplicate_profile_id}")
   end
 
-  def prevent_same_school_different_lead_provider
-    return unless primary_profile.lead_provider&.id != duplicate_profile.lead_provider&.id && primary_profile.school == duplicate_profile.school
+  def reconcile?
+    !same_school_different_lead_provider? || duplicate_has_declarations?
```

Comment:
Im wondering around the second check for declarations, so let's say the duplicate/primary have the same lead provider, but different schools, and duplicate has no declarations, I think we'll probably want to run through some steps, unless I misunderstood the check 🤔

Reply:
I thought if we had the same school but different lead provider and no declarations on the duplicate then we wanted to merge without reconciling any data? and if we have the same school but different lead providers and the duplicate has declarations it will raise an error. Have I misunderstood?